app/controllers/casa_cases_controller.rb

```diff
@@ -157,17 +159,15 @@ def casa_case_params
       :court_report_due_date,
       :empty_court_date,
       contact_type_ids: [],
-      court_dates_attributes: [:date]
+      court_dates_attributes: [:date],
+      case_assignments_attributes: [:volunteer_id]
     )
   end
 
   def casa_case_create_params
-    create_params = if court_date_unknown?
-      casa_case_params.except(:court_dates_attributes)
-    else
-      casa_case_params
-    end
-
+    create_params = casa_case_params
+    create_params = create_params.except(:court_dates_attributes) if court_date_unknown?
+    create_params = create_params.except(:case_assignments_attributes) unless assigned_volunteer?
```

Comment:
It might be a symptom of another issue but if you don't exclude these attributes (case assignment and court date), and they are left blank on the form, there will be an error.

<img width="685" alt="image" src="https://github.com/rubyforgood/casa/assets/15682136/171488e0-735e-49ed-bc84-07bc64fd421b">

Though I think this can be fixed by updating the model to do something like this

```rb
class CasaCase
# ...
  accepts_nested_attributes_for :case_assignments, reject_if: proc { |attributes| attributes['volunteer_id'].blank? }
# ...
end
```

I grabbed this from here: https://stackoverflow.com/questions/56096562/how-to-do-optional-nested-forms-with-form-for-in-rails-5